### PR TITLE
Allow operators to setup a default certificate

### PR DESCRIPTION
**Pull request by dprotaso**
Fixes: https://github.com/knative-sandbox/net-contour/issues/182

### CODE REVIEW

**Comment by dprotaso** [comment]
/hold

going to stage the docs

File: pkg/reconciler/contour/config/contour.go [hunk 1]
```diff
@@ -31,14 +31,16 @@ const (
...
 )
 
 // Contour contains contour related configuration defined in the
 // contour config map.
 type Contour struct {
-	VisibilityKeys    map[v1alpha1.IngressVisibility]sets.String
-	VisibilityClasses map[v1alpha1.IngressVisibility]string
+	VisibilityKeys       map[v1alpha1.IngressVisibility]sets.String
+	VisibilityClasses    map[v1alpha1.IngressVisibility]string
+	DefaultTLSSecretName string
```
Comment (mattmoor) [inline pkg/reconciler/contour/config/contour.go:43]: `types.NamespacedName`?

File: pkg/reconciler/contour/config/contour.go [hunk 2]
```diff
@@ -48,10 +50,13 @@ type visibilityValue struct {
 
 // NewContourFromConfigMap creates an Contour config from the supplied ConfigMap
 func NewContourFromConfigMap(configMap *corev1.ConfigMap) (*Contour, error) {
+	defaultSecret := configMap.Data[defaultTLSSecretConfigKey]
```
Comment (mattmoor) [inline pkg/reconciler/contour/config/contour.go:53]: Validate the structure?

File: pkg/reconciler/contour/config/contour.go (hunk 1 above)
Comment (dprotaso) [reply pkg/reconciler/contour/config/contour.go:43]: Yeah we should fully qualify it with the namespace

**Comment by dprotaso** [comment]
rebased

**Pull request merged**